propagation/ondemandservices.qmd

```diff
@@ -49,15 +51,16 @@ For further guidance on creating and using UDPs, you can explore the following r
 * [Jupyter notebook example](https://github.com/Open-EO/openeo-community-examples/blob/main/python/Sentinel1_Stats/Sentinel1_Stats.ipynb)
 * [Official openEO specification](https://api.openeo.org/#tag/User-Defined-Processes)
 
-### Building an OGC Application Package based service
+### Building an EOAP Process based service
+
+Another option is to package your software using Docker containers. This approach, based on the OGC Best Practice for Earth Observation Application Packaging (EOAP), is very generic and is particularly relevant if you have an existing piece of complex software that you want to make available as a service.
 
-Another option is to package your software using Docker containers. This approach is very generic and is particularly
-relevant if you have an existing piece of complex software that you want to make available as a service.
+This APEx compliant, cloud service implementation option, consists in containerizing your software (using Docker technologies) along with CWL wrappers (Common Workflow Language)[@cwl], together forming a reusable, scalable Earth Observation Application Package (EOAP) [@ap]. APEx can provide the guidance and support you need to efficiently translate your algorithm into an EOAP implementation.
 
-Another APEx compliant cloud service implementation option is to package your software using Docker containers and CWL (
-Common Workflow Language)[@cwl] wrappers, together forming an OGC Application Package [@ap]. This approach is very
-generic and is particularly relevant if you have an existing piece of complex software that you want to make available
-as a service.
+After your EOAP is created, it can be registered in
```

Comment:
Shouldn't the EOAP first be deployed and tested as an OGC API Process before projects can register it in the catalogue? The current wording might give the impression that the EOAP alone is sufficient for registration. While the sentence mentions that it must be "associated with one or more backends or platforms," it does not clearly emphasize that this is a prerequisite.

Suggestion for the text:

After creating your EOAP, it must first be deployed and validated as an OGC API Process on an APEx-compatible backend or platform. After its successful deployment and testing can it be registered in the [APEx Algorithm Services ...]

Reply:
yes, OK for the testing & validation part.
On the wording, please stop associating an EOAP with an API.
An EOAP compliant software embeds more APIs than only OGC API Process.

Both the OGC API Process and the openEO API (for the parts related to processing job operations) are covering the APEx functional requirements of exposing a capitalised algorithm "as-a-Service": the job parameters definition and the job run management are defined programatically, for Machine-to-Machine interoperability.

This level on interoperability serves:
- the use of authorized CLI clients (batch requests, ...)
- the use of embedded Web clients (i.e. integrated on the host platform to support user communities)